pkg/command/client.go

```diff
@@ -265,6 +266,211 @@ func (c *Client) updatePanelTargets(queryManager *QueryManager, panel *simplejso
+	}
+	return queryPaths
+}
+
+func (c *Client) UpdateDashboardDescriptions(ctx context.Context, uid string, overwrite bool, dryRun bool) error {
+	dashboardFull, err := c.GetDashboardByUID(ctx, uid)
+	if err != nil {
+		return err
+	}
+
+	dashboardTitle := dashboardFull.Dashboard.Get("title").MustString()
+	if dashboardTitle == "" {
+		return fmt.Errorf("dashboard has no title")
+	}
+
+	// Get folder title from metadata
+	folderTitle := dashboardFull.Meta.Get("folderTitle").MustString()
+
+	c.logd("processing dashboard: %s in folder: %s (overwrite: %v, dryRun: %v)", dashboardTitle, folderTitle, overwrite, dryRun)
+
+	panelsUpdated := 0
+	panelsSkipped := 0
+
+	// Get all panels
+	panels := dashboardFull.Dashboard.Get("panels").MustArray()
+	c.logd("found %d panels to process", len(panels))
+
+	// Process all panels
+	for _, panelBy := range panels {
+		panel := simplejson.NewFromAny(panelBy)
+		panelType := panel.Get("type").MustString()
+		if panelType == "row" {
+			rowTitle := panel.Get("title").MustString()
+			for _, subPanelBy := range panel.Get("panels").MustArray() {
+				subPanel := simplejson.NewFromAny(subPanelBy)
+				updated, skipped, err := c.updatePanelDescription(subPanel, folderTitle, dashboardTitle, rowTitle, overwrite, dryRun)
+				if err != nil {
+					return err
+				}
+				if updated {
+					panelsUpdated++
+				}
+				if skipped {
+					panelsSkipped++
+				}
+			}
+		} else {
+			updated, skipped, err := c.updatePanelDescription(panel, folderTitle, dashboardTitle, "", overwrite, dryRun)
+			if err != nil {
+				return err
+			}
+			if updated {
+				panelsUpdated++
+			}
+			if skipped {
+				panelsSkipped++
+			}
+		}
+	}
+
+	if dryRun {
+		c.logd("DRY RUN: would update %d panels, skip %d panels", panelsUpdated, panelsSkipped)
+	} else {
+		// Save the updated dashboard
+		if err := c.SaveDashboard(ctx, &grafsdk.DashboardSavePayload{
+			Dashboard: dashboardFull.Dashboard,
+			Overwrite: true,
+			FolderID:  dashboardFull.Meta.Get("folderId").MustInt64(),
+		}); err != nil {
+			return fmt.Errorf("SaveDashboard: %w", err)
+		}
+		c.logd("updated %d panels, skipped %d panels", panelsUpdated, panelsSkipped)
+	}
+
+	return nil
+}
+
+func (c *Client) updatePanelDescription(panel *simplejson.Json, folderTitle, dashboardTitle, rowTitle string, overwrite, dryRun bool) (bool, bool, error) {
+	panelType := panel.Get("type").MustString()
+	panelTitle := panel.Get("title").MustString()
+	currentDesc := panel.Get("description").MustString()
+
+	// Check if we should update this panel
+	shouldUpdate := overwrite || c.isInvalidDescription(currentDesc)
+	if !shouldUpdate {
+		return false, true, nil
+	}
+
+	// Generate new description
+	newDesc := c.generatePanelDescription(folderTitle, dashboardTitle, rowTitle, panelType, panelTitle)
+	if newDesc == currentDesc {
+		return false, true, nil
+	}
+
+	// Update the panel description
+	if !dryRun {
+		panel.Set("description", newDesc)
+	}
+
+	action := "would update"
+	if !dryRun {
```

Comment:
suggestion: merge all the dry run statements in a single condition, for example:
```
if !dryRun {
  panel.Set("description", newDesc)
  action = "updated"
}